lib/charms/opensearch/v0/opensearch_plugin_manager.py

```diff
@@ -340,28 +420,30 @@ def status(self, plugin: OpenSearchPlugin) -> PluginState:
 
         # The _user_request_to_enable comes first, as it ensures there is a relation/config
         # set, which will be used by _is_enabled to determine if we are enabled or not.
-        if not self._user_requested_to_enable(plugin) and not self._is_enabled(plugin):
-            return PluginState.DISABLED
-
-        if self._is_enabled(plugin):
-            return PluginState.ENABLED
-
+        try:
+            if not self._is_enabled(plugin):
+                return PluginState.DISABLED
+            elif self._user_requested_to_enable(plugin):
+                return PluginState.ENABLED
```

Comment:
i believe here we're mixing 2 different logics, 1 to check whether the user requested to enable and the other whether it's actually enabled

Reply:
That `elif` means: `self._is_enabled(plugin)==True` and the user still wants to keep that plugin - `self._user_requested_to_enable(plugin)==True`.

So, the full matrix would be:
```
self._is_enabled(plugin)==False and self._user_requested_to_enable(plugin)==<does not matter>
    return DISABLED

self._is_enabled(plugin)==True and self._user_requested_to_enable(plugin)==True
    return ENABLED

self._is_enabled(plugin)==True and self._user_requested_to_enable(plugin)==False
    return INSTALLED
```

Reply:
We will add another state: `ENABLING`, which represents:
```
self._is_enabled(plugin)==False and self._user_requested_to_enable(plugin)==False
    return DISABLED
self._is_enabled(plugin)==False and self._user_requested_to_enable(plugin)==True
    return ENABLING_NEEDED
self._is_enabled(plugin)==True and self._user_requested_to_enable(plugin)==True
    return ENABLED
self._is_enabled(plugin)==True and self._user_requested_to_enable(plugin)==False
    return DISABLING_NEEDED
```